### PR TITLE
Remove strict qc controls for uploading microsalt cases to HK

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
In #1655 a qc control was implemented for microSALT cases, which caused some complete cases not to update their status in statusDB from "running". This caused the files never to be uploaded to Housekeeper. As suggested in #2500, the qc controls were removed, and all the complete cases are returned when `get_cases_to_store` is called. The function `get_cases_to_store` in MicrosaltAPI, which overrode the class from AnalysisAPI, was removed, using the class from the base class with the intended behaviour.

### Added
- A test function for `get_cases_to_store ` when called through the MicrosaltAPI

### Fixed

- Removed functions `is_qc_required`, `get_completed_cases`, and `get_cases_to_store `. This last one is implemented in the base class with the desired functionality
- Remove tests for old `get_cases_to_store`

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b fix-microsalt-upload-hk -a
    ```

### How to test

- [x] `cg workflow microsalt store-available --dry-run`

### Expected test outcome

- [x] Check that the cases to be stored are running in statusDB

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO, VJ
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Deploy this branch on 
  - [ ] cg stage
  - [ ] cg production

### CODE REVIEW

File: cg/meta/workflow/microsalt.py [hunk 1]
```diff
@@ -237,54 +235,6 @@ def get_project(self, sample_id: str) -> str:
         """Get LIMS project for a sample"""
         return self.lims_api.get_sample_project(sample_id)
 
-    def get_cases_to_store(self) -> List[Family]:
```
Comment (diitaz93) [inline cg/meta/workflow/microsalt.py:240]: The function is implemented in the `AnalysisAPI` base class as:
```python
    def get_cases_to_store(self) -> List[Family]:
        """Return cases where analysis finished successfully,
        and is ready to be stored in Housekeeper."""
        return [
            case
            for case in self.status_db.get_running_cases_in_pipeline(pipeline=self.pipeline)
            if self.trailblazer_api.is_latest_analysis_completed(case_id=case.internal_id)
        ]
```

**Review by ChrOertlin — APPROVED** [review]
👍  Looks good

**Comment by diitaz93** [comment]
## Tests on stage
```shell
[js.diazboada@hasta:~] [S_base] 17s $ cg workflow microsalt store-available --dry-run
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Storing deliverables for dearsheep
Case dearsheep exists in Status DB
Dry run: Would have stored deliverables for dearsheep
Storing deliverables for usefulswift
Case usefulswift exists in Status DB
Dry run: Would have stored deliverables for usefulswift
Storing deliverables for suitedcolt
Case suitedcolt exists in Status DB
Dry run: Would have stored deliverables for suitedcolt
Storing deliverables for drivencrow
Case drivencrow exists in Status DB
Dry run: Would have stored deliverables for drivencrow
Storing deliverables for coolviper
Case coolviper exists in Status DB
Dry run: Would have stored deliverables for coolviper
Storing deliverables for safehyena
Case safehyena exists in Status DB
Dry run: Would have stored deliverables for safehyena
Storing deliverables for rapidrhino
Case rapidrhino exists in Status DB
Dry run: Would have stored deliverables for rapidrhino
θ61° [js.diazboada@hasta:~] [S_base] 19s $ cg get case dearsheep --no-samples --no-analyses --no-relate
+-----------+--------+------------+------------+----------+----------+
| Family    |   Name | Customer   | Priority   | Panels   | Action   |
|-----------+--------+------------+------------+----------+----------|
| dearsheep | XXXX | XXXX    | standard   |          | running  |
+-----------+--------+------------+------------+----------+----------+
θ61° [js.diazboada@hasta:~] [S_base] 19s $ cg get case usefulswift --no-samples --no-analyses --no-relate
+-------------+--------+------------+------------+----------+----------+
| Family      |   Name | Customer   | Priority   | Panels   | Action   |
|-------------+--------+------------+------------+----------+----------|
| usefulswift | XXXX | XXXX    | priority   |          | running  |
+-------------+--------+------------+------------+----------+----------+
θ62° [js.diazboada@hasta:~] [S_base] 18s $ cg get case suitedcolt --no-samples --no-analyses --no-relate
+------------+--------+------------+------------+----------+----------+
| Family     |   Name | Customer   | Priority   | Panels   | Action   |
|------------+--------+------------+------------+----------+----------|
| suitedcolt | XXXX | XXXX    | priority   |          | running  |
+------------+--------+------------+------------+----------+----------+
θ60° [js.diazboada@hasta:~] [S_base] 17s $ cg get case drivencrow --no-samples --no-analyses --no-relate
+------------+--------+------------+------------+----------+----------+
| Family     |   Name | Customer   | Priority   | Panels   | Action   |
|------------+--------+------------+------------+----------+----------|
| drivencrow | XXXX | XXXX    | priority   |          | running  |
+------------+--------+------------+------------+----------+----------+
θ61° [js.diazboada@hasta:~] [S_base] 19s $ cg get case coolviper --no-samples --no-analyses --no-relate
+-----------+--------+------------+------------+----------+----------+
| Family    |   Name | Customer   | Priority   | Panels   | Action   |
|-----------+--------+------------+------------+----------+----------|
| coolviper | XXXX | XXXX    | research   |          | running  |
+-----------+--------+------------+------------+----------+----------+
θ63° [js.diazboada@hasta:~] [S_base] 19s $ cg get case safehyena --no-samples --no-analyses --no-relate
+-----------+--------+------------+------------+----------+----------+
| Family    |   Name | Customer   | Priority   | Panels   | Action   |
|-----------+--------+------------+------------+----------+----------|
| safehyena | XXXX | XXXX    | priority   |          | running  |
+-----------+--------+------------+------------+----------+----------+
θ62° [js.diazboada@hasta:~] [S_base] 18s $ cg get case rapidrhino --no-samples --no-analyses --no-relate
+------------+--------+------------+------------+----------+----------+
| Family     |   Name | Customer   | Priority   | Panels   | Action   |
|------------+--------+------------+------------+----------+----------|
| rapidrhino | XXXX | XXXX    | priority   |          | running  |
+------------+--------+------------+------------+----------+----------+
θ62° [js.diazboada@hasta:~] [S_base] 18s $ 
```

**Review by Vince-janv — APPROVED** [review]
Looks good 👍  I think the following should be tested. Start a case, let it finish, then call store-available and see that it gets added in housekeeper

**Comment by diitaz93** [comment]
## Test passed on stage
```shell
Storing deliverables for coolviper
Case coolviper exists in Status DB
Storing bundle data in Housekeeper for coolviper
Found deliverables file /home/proj/stage/microbial/results/reports/deliverables/743867_deliverables.yaml
Found deliverables file /home/proj/stage/microbial/results/reports/deliverables/743867_deliverables.yaml
Converting pipeline deliverables to CG deliverables
Running command /home/proj/stage/bin/miniconda3/envs/S_hermes/bin/hermes convert deliverables --pipeline microsalt /home/proj/stage/microbial/results/reports/deliverables/743867_deliverables.yaml
Fetching bundle with name: coolviper
Created new bundle: coolviper
```

**Comment by diitaz93** [comment]
## Deployed to stage:
```shell
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/js.diazboada
done.
Log deploy... done.
cg, version 49.5.3

```
## Deployed to production:
```shell
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/js.diazboada
done.
Log deploy... done.
cg, version 49.5.3
```

**Pull request merged**